core/platform/auth/firebase_auth_services.py

```diff
@@ -136,6 +128,40 @@ def authenticate_request(request):
     return None if not auth_id else auth_domain.AuthClaims(auth_id, email)
 
 
+def delete_associations(user_id):
+    """Deletes associations referring to the given user_id.
+
+    Args:
+        user_id: str. The ID of the user being deleted.
+
+    Returns:
+        bool. Whether the user's data has been successfully deleted.
+    """
+    assoc = auth_models.UserIdByFirebaseAuthIdModel.get_by_user_id(user_id)
```

Comment:
Name this {{something}}_model -- this is our convention for datastore model instances.